### PR TITLE
Add composite key support in Phoenix coder

### PR DESCRIPTION
**Pull request by weiqingy**
## What changes were proposed in this pull request?

This PR is to add composite key support in Phoenix coder.

## How was this patch tested?

Passed current unit tests and added new unit tests.

### CODE REVIEW

**Comment by weiqingy** [comment]
This PR used`ImmutableBytesWritable` and `RowKeySchema` from Phoenix directly. To make the PR simple, I did not copy the related code into SHC (we may do this later).

P.S. the commit which adds phoenix coder for the cases of non-composite key is [here](https://github.com/hortonworks-spark/shc/pull/79/commits/a74a8f3937e4d1b1d278fdd8e95c7149d5b5becb) (committed in [PR#79](https://github.com/hortonworks-spark/shc/pull/79)) 

Hi, @enis @dongjoon-hyun Could you please help to review this PR? Thanks.

**Comment by dongjoon-hyun** [comment]
+1 LGTM. @weiqingy .

**Comment by weiqingy** [comment]
Thanks for the review, @dongjoon-hyun .

File: core/src/main/scala/org/apache/spark/sql/execution/datasources/hbase/types/Phoenix.scala [hunk 1]
```diff
@@ -57,6 +57,41 @@ class Phoenix(f:Option[Field] = None) extends SHCDataType {
...
+
+  override def decodeCompositeRowKey(row: Array[Byte], keyFields: Seq[Field]): Map[Field, Any] = {
+    def buildSchema(): RowKeySchema = {
+      val builder: RowKeySchemaBuilder = new RowKeySchemaBuilder(keyFields.length)
+      keyFields.foreach{ x =>
+        builder.addField(buildPDatum(x.dt), false, SortOrder.getDefault)
+      }
+      builder.build
+    }
+    val schema: RowKeySchema = buildSchema()
```
Comment (enis) [inline core/src/main/scala/org/apache/spark/sql/execution/datasources/hbase/types/Phoenix.scala]: This should happen once for task, not once per row right?

Comment (weiqingy) [inline core/src/main/scala/org/apache/spark/sql/execution/datasources/hbase/types/Phoenix.scala]: Thanks for the review, @enis. Yes. I have submitted [PR#87](https://github.com/hortonworks-spark/shc/pull/87). Could you please help to review it? Thanks.

**Pull request merged**